src/map/skill.c

```diff
@@ -7412,21 +7411,22 @@ int skill_castend_nodamage_id(struct block_list *src, struct block_list *bl, uin
 					return 1;
 				}
 				if( sd ) {
-					int x,bonus=100, potion = min(500+skill_lv,505);
-					x = skill_lv%11 - 1;
-					i = pc->search_inventory(sd,skill->dbs->db[skill_id].itemid[x]);
-					if (i == INDEX_NOT_FOUND || skill->dbs->db[skill_id].itemid[x] <= 0) {
+					int bonus = 100, potion = min(500+skill_lv,505);
+					int item_idx = skill_lv%11 - 1;
```

Comment:
probably 11 better replace to MAX_SKILL_ITEM_REQUIRE + 1?

Reply:
Hmm good point. This was probably be meant to be `(skill_lv - 1) % MAX_SKILL_ITEM_REQUIRE` (which correctly ranges between `0` and `MAX_SKILL_ITEM_REQUIRE`). Right now it can even return `-1`...